Task3/Task_3_Bogdan_Topalov/car_wash_db_pool/src/python/car_wash/db/db_pool.py

```diff
@@ -0,0 +1,116 @@
+import time
+from contextlib import contextmanager
+
+import psycopg2
+
+from ..utils import Singleton
+
+
+DB_CONFIG = {
+    'host': 'localhost',
+    'port': '5432',
+    'db_name': 'carwashDB',
+    'username': 'bogdan',
+    'password': '123456'
+}
+
+
+class DBPoolBase(metaclass=Singleton):
+
+    """Base class of DB pool."""
+
+    def _create_connection(self) -> psycopg2.connect:
+        """Create connection to DB"""
+        raise NotImplementedError('No code added into _create_connection method')
+
+
+class RODBPool(DBPoolBase):
+
+    """Read only pool of connections."""
+
+    def __init__(self, hostname: str, password: str, username: str, _dbname: str, pool_size: int) -> None:
+        """Constructor of RODBPool connections
+
+        :Parameters:
+            - `hostname` - DB hostname
+        """
+        super().__init__()
+        self._pool = []
+        self._hostname = hostname
+        self._password = password
+        self._username = username
+        self._dbname = _dbname
+        self._pool_size = pool_size
+
+    def _create_connection(self) -> psycopg2.connect:
+        """Create connection to DB"""
+        return psycopg2.connect(dbname=self._dbname,
+                                user=self._username,
+                                password=self._password,
+                                host=self._hostname)
+
+    @contextmanager
+    def connection(self) -> psycopg2.connect:
```

Comment:
Please check the comments: https://github.com/mehalyna/CarWashing-System/pull/38/files#r928253372
https://github.com/mehalyna/CarWashing-System/pull/38/files#r928255340